src/components/publishing/share.tsx

```diff
@@ -7,29 +8,48 @@ interface ShareProps extends React.HTMLProps<HTMLDivElement> {
   title: string
 }
 
-const Share: React.SFC<ShareProps> = props => {
-  const { url, title } = props
-  const encodedUrl = encodeURIComponent(url)
-  const getHref = type => {
+@track()
+class Share extends React.Component<ShareProps, null> {
+  constructor(props) {
+    super(props)
+    this.getHref = this.getHref.bind(this)
+    this.trackShare = this.trackShare.bind(this)
```

Comment:
We can probably get rid of these by defining the functions with ES6 arrows: `trackShare = (e) => {}`

Reply:
This should work, because react-tracking uses this also. Fun fact, in React Native / Typescript, this actually gives issues when decorating functions that use this approach. If anyone looked at [my PR](https://github.com/artsy/emission/pull/693) and noted the second "Important" point, it's specific to Emission's setup.

Reply:
Wait, forget what I just said. Test it, but it might also give issues (like I had in Emission) because obviously Reaction is also using Typescript.

Reply:
This feature (specifically ES7 class properties, not ES6 arrow functions) has only [very recently](https://github.com/tc39/proposal-class-fields/commit/668b51adc7b62a98f81f49b122302d3fbd9d6f06) moved to stage 3 of TC39, which is why it was dangerous to use (implementations could differ, debuggers may not be able to make sense of the emitted code, etc). Now that it has moved to stage 3, the TypeScript implementers will probably want to hear about cases where it doesn’t work well, so let’s verify on the next TS update and otherwise submit a bug report.